app/src/main/java/org/eyeseetea/malariacare/data/remote/drive/DriveRestController.java

```diff
@@ -0,0 +1,192 @@
+package org.eyeseetea.malariacare.data.remote.drive;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.google.android.gms.common.GooglePlayServicesUtil;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.services.drive.DriveScopes;
+
+import org.eyeseetea.malariacare.DashboardActivity;
+import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.data.database.model.MediaDB;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.data.repositories.MediaRepository;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+
+public class DriveRestController {
```

Comment:
this class seems a DownloadMediaUseCase, can you refactor this class to use case respecting clean architecture rules abstracting external dependencies?